src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VsCommonOperations.cs

```diff
@@ -14,43 +15,33 @@ namespace NuGet.PackageManagement.VisualStudio
     [Export(typeof(ICommonOperations))]
     internal sealed class VsCommonOperations : ICommonOperations
     {
-        private readonly Lazy<EnvDTE.DTE> _dte;
         private IDictionary<string, ISet<VsHierarchyItem>> _expandedNodes;
+        private readonly IAsyncServiceProvider _asyncServiceProvider;
 
         [ImportingConstructor]
         public VsCommonOperations(
             [Import(typeof(SVsServiceProvider))]
-            IServiceProvider serviceProvider)
+            IAsyncServiceProvider asyncServiceProvider)
         {
-            if (serviceProvider == null)
-            {
-                throw new ArgumentNullException(nameof(serviceProvider));
-            }
-
-            _dte = new Lazy<EnvDTE.DTE>(
```

Comment:
Given the scenario, It might not matter too much, but why did you remove the caching of the DTE object here?

Reply:
Sorry, not intentional. Replaced with AsyncLazy, please check if implementation is correct.